aws/src/main/java/org/apache/iceberg/aws/s3/PrefixedS3Client.java

```diff
@@ -50,12 +53,12 @@ class PrefixedS3Client implements AutoCloseable {
     this.s3FileIOProperties = new S3FileIOProperties(properties);
     // Do not override s3 client if it was provided
     if (s3 == null) {
-      Object clientFactory = S3FileIOAwsClientFactories.initialize(properties);
-      if (clientFactory instanceof S3FileIOAwsClientFactory) {
-        this.s3 = ((S3FileIOAwsClientFactory) clientFactory)::s3;
-      }
-      if (clientFactory instanceof AwsClientFactory) {
-        this.s3 = ((AwsClientFactory) clientFactory)::s3;
+      Object factory = S3FileIOAwsClientFactories.initialize(properties);
+      if (factory instanceof S3FileIOAwsClientFactory) {
+        this.clientFactory = (S3FileIOAwsClientFactory) factory;
+        this.s3 = clientFactory::s3;
+      } else if (factory instanceof AwsClientFactory) {
+        this.s3 = ((AwsClientFactory) factory)::s3;
```

Comment:
Minor: Could we undo this naming change and refactoring, and the one right below for async. It makes the diff larger than needed, and reviewers end up having to analyze these lines to make sure there's nothing incorrect.